lib/Transforms/FuncSSAToGSA.cpp

```diff
@@ -0,0 +1,45 @@
+#include "dynamatic/Transforms/FuncSSAToGSA.h"
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "llvm/ADT/STLExtras.h"
+
+#include "dynamatic/Analysis/ControlDependenceAnalysis.h"
+
+using namespace mlir;
+using namespace dynamatic;
+
+namespace {
+
+// Simple driver for the pass. 
+struct FuncSSAToGSAPass
+    : public dynamatic::impl::FuncSSAToGSABase<FuncSSAToGSAPass> {
+
+  public:
+    void runOnOperation() override {
+      translate_ssa_to_gsa(getOperation());
+    };
+  public:
+    void translate_ssa_to_gsa(func::FuncOp funcOp);  // the main function of the pass
+  };
+}; // namespace
+
+void FuncSSAToGSAPass::translate_ssa_to_gsa(func::FuncOp funcOp) {
```

Comment:
Invalid case: here it should be "translateSSAToGSA"

https://github.com/EPFL-LAP/dynamatic/blob/main/docs/GettingStarted.md#contributing